internal/v4/auth.go

```diff
@@ -65,10 +69,11 @@ func (h *ReqHandler) authorize(req *http.Request, acl []string, alwaysAuth bool,
 	}
 
 	// Macaroon verification failed: mint a new macaroon.
-	m, err := h.newMacaroon()
+	m, err := h.newMacaroon(checkers.DenyCaveat(opReadArchive))
```

Comment:
// We need to deny access for opReadArchive operations because they
// may require more specific checks that terms and conditions have been
// satisfied.
?